brainscore_vision/nwb_integration/generate_data_files.py

```diff
@@ -0,0 +1,201 @@
+from pathlib import Path
+from dandi_to_stimulus_set import get_stimuli
+from extract_nwb_data import validate_nwb_file
+from create_assembly import load_responses
+from brainscore_vision.model_helpers.utils import fullname
+import os, json
+import logging
+import zipfile
+
+
+class DataFactory:
+    def __init__(self, user_json: str):
+        self._logger = logging.getLogger(fullname(self))
+
+        self.user_json = json.loads(user_json)
+        self.parse_json()
+
+        self.nwb_file   = validate_nwb_file(self.nwb_file_path, dandiset_id=self.dandiset_id)
+        self.json_file_path = os.path.join(self.exp_path, "nwb_metadata.json")
+        self.generate_json_file()
+        with open(self.json_file_path, 'r') as f:
+            self.params = json.load(f)
+
+    def __call__(self):
+        try: os.mkdir(os.path.join(self.directory, 'test_data_packaging'))
+        except: pass 
+
+        output_dir = os.path.join(self.directory, 'test_data_packaging')
+        os.mkdir(os.path.join(output_dir, 'data'))
+        output_dir = os.path.join(output_dir, 'data')
+        os.mkdir(os.path.join(output_dir, 'data_packaging'))
+        output_dir = os.path.join(output_dir, 'data_packaging')
+
+        # create and write data packaging code:
+        data_packaging_code = self.generate_data_packaging_code()
+        data_packaging_path = Path(f"{output_dir}/data_packaging.py")
+        self.write_code_into_file(data_packaging_code, data_packaging_path)
+        self._logger.debug('Finished writing data_packaging.py')
+
+        # create and write init code:
+        init_code = self.generate_init_code()
+        init_path = Path(f"{output_dir}/__init__.py")
+        self.write_code_into_file(init_code, init_path)
+        self._logger.debug('Finished writing __init__.py')
+
+        # create and write test code:
+        test_code = self.generate_test_code()
+        test_path = Path(f"{output_dir}/test.py")
+        self.write_code_into_file(test_code, test_path)
+        self._logger.debug('Finished writing test.py')
+
+        os.system(f'mv test_data_packaging/data/data_packaging test_data_packaging/data/{self.identifier}')
```

Comment:
Is there any way we can make this safer for a scenario where code is called from a different directory?